proxy.go

```diff
@@ -113,6 +113,16 @@ func (p *Proxy) SetMITM(config *mitm.Config) {
 	p.mitm = config
 }
 
+// SetSkipTLSVerify configures whether the proxy skips verification of server certificates.
+func (p *Proxy) SetSkipTLSVerify(skip bool) {
```

Comment:
Should I replicate all the timeouts from NewProxy()?  If so, do you want them to be flags or just hard code it?